src/components/form/HdRange.vue

```diff
@@ -0,0 +1,320 @@
+<template>
+  <div
+    :class="fieldClasses"
+    class="range field"
+  >
+    <input type="range"
+      ref="input"
+      v-model="currentValue"
+      :id="name"
+      :name="name"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="handleChange"
+      @focus="handleFocus"
+      @blur="handleBlur"
+    >
+    <div class="range__thumb" ref="thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress" ref="progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+    isLegacy: {
+      type: Boolean,
+      default: false,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      isActive: null,
+      currentValue: this.value,
+    };
+  },
+  computed: {
+    fieldClasses() {
+      return {
+        'field--active': this.isActive,
+        'field--disabled': this.disabled,
+      };
+    },
+  },
+  watch: {
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+    rangeStep() {
+      this.updateRangeDecoration();
+    },
+    value() {
+      this.currentValue = this.value;
+      this.updateRangeDecoration();
+    },
+  },
+  mounted() {
+    this.progress = this.$refs.progress;
+    this.progress.style.width = '1px';
+
+    this.thumb = this.$refs.thumb;
+
+    this.updateRangeDecoration();
+  },
+  methods: {
+    normalizeAndEmit() {
+      if (this.currentValue > this.maxValue) {
+        this.currentValue = this.maxValue;
+      } else if (this.currentValue < this.minValue) {
+        this.currentValue = this.minValue;
+      }
+
+      this.currentValue = this.currentValue - (this.currentValue % this.rangeStep);
+
+      this.$emit('input', parseFloat(this.currentValue));
```

Comment:
Small optional thing:
You don't have to parse the value to float yourself you can use [`v-modal.number`](https://vuejs.org/v2/guide/forms.html#number) (it does exactly the same thing, I only find it cleaner, and it's always nice to parse as early as possible).

Reply:
Very nice!